### PR TITLE
Tree component for new Permissions Page

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

I started extracting individual components for [the Permissions Page revamping](https://www.notion.so/Let-the-Permissions-page-scale-51a4a21940924ab0b841e8b772d0237c) to make review easier.
This PR contains purely UI tree component. Please, ignore the rest, I just needed to render it on some page.

It is temporarily rendered on http://localhost:3000/admin/permissions/new/databases

### Screenshot
Screen Shot 2021-06-09 at 00 46 51

### CODE REVIEW

File: frontend/src/metabase/admin/permissions/components/tree/TreeNode.styled.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  background-color: ${props =>
+    props.isSelected ? colors["accent7"] : "unset"};
+  padding-left: ${props => props.depth + 1}rem;
+  padding-right: 0.5rem;
+  cursor: pointer;
+  font-weight: 700;
+
+  &:hover {
+    background-color: ${props =>
+      props.isSelected ? colors["accent7"] : `#e7e6f1`};
```
Comment (daltojohnso) [inline frontend/src/metabase/admin/permissions/components/tree/TreeNode.styled.jsx:19]: should we put this color somewhere? fwiw I've also seen code like `lighten(colors("accent7"))` that might give you a similar color? not sure.

File: frontend/src/metabase/admin/permissions/components/tree/TreeNode.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    switch (key) {
+      case "Enter":
+        onSelect(id);
+        break;
+      case "ArrowRight":
+        !isExpanded && onToggleExpand(id);
+        break;
+      case "ArrowLeft":
+        isExpanded && onToggleExpand(id);
+        break;
```
Comment (daltojohnso) [inline frontend/src/metabase/admin/permissions/components/tree/TreeNode.jsx:57]: nice a11y -- I wonder how difficult it'd be to also implement an `ArrowUp` and `ArrowDown` action. I was playing with the page using my keyboard and it felt like it'd be a nice thing to have. No need to do it in this PR, though.

**Review by daltojohnso — APPROVED** [review]
nice & clean. good work.

File: frontend/src/metabase/admin/permissions/components/tree/TreeNode.styled.jsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/admin/permissions/components/tree/TreeNode.styled.jsx]: Thanks for the suggestion, I changed that

File: frontend/src/metabase/admin/permissions/components/tree/TreeNode.jsx (hunk 2 above)
Comment (alxnddr) [reply frontend/src/metabase/admin/permissions/components/tree/TreeNode.jsx:57]: Totally agree, I'll work on it separately

**Pull request merged**